packages/frontend/src/components/RefreshButton.tsx

```diff
@@ -1,28 +1,23 @@
 import { Button } from '@blueprintjs/core';
 import React from 'react';
-import { UseQueryResult } from 'react-query';
-import { ApiError, ApiQueryResults } from 'common';
 import { useExplorer } from '../providers/ExplorerProvider';
 import { useTracking } from '../providers/TrackingProvider';
 import { EventName } from '../types/Events';
 import { useQueryResults } from '../hooks/useQueryResults';
 
-type RefreshButtonProps = {
-    queryResults: UseQueryResult<ApiQueryResults, ApiError>;
-};
 export const RefreshButton = () => {
     const {
         state: { isValidQuery },
+        actions: { syncState },
     } = useExplorer();
-    const queryResults = useQueryResults(false);
-    const { refetch, isFetching } = queryResults;
+    const { isFetching } = useQueryResults();
     const { track } = useTracking();
     return (
         <Button
             intent="primary"
             style={{ height: '40px', width: 150, marginRight: '10px' }}
             onClick={() => {
-                refetch();
+                syncState();
```

Comment:
Users wont be able to rerun the same query, since react query uses key string to compare. Refetch fits the purpose well as the user is trying to fetch/refetch the query.

Reply:
Good point, changing it so when a user clicks the run query button it refetch the data.